src/ui/hooks/useBiometricsHook.ts

```diff
@@ -0,0 +1,109 @@
+import { useState, useEffect } from "react";
+import {
+  BiometricAuth,
+  BiometryError,
+  BiometryErrorType,
+} from "@aparajita/capacitor-biometric-auth";
+import {
+  AndroidBiometryStrength,
+  CheckBiometryResult,
+} from "@aparajita/capacitor-biometric-auth/dist/esm/definitions";
+import i18n from "i18next";
+import { PluginListenerHandle } from "@capacitor/core";
+import { useActivityTimer } from "../components/AppWrapper/hooks/useActivityTimer";
+import { PreferencesKeys, PreferencesStorage } from "../../core/storage";
+
+const useBiometricAuth = () => {
+  const [biometricInfo, setBiometricInfo] = useState<CheckBiometryResult>();
+  const [biometricsIsEnabled, setBiometricsIsEnabled] = useState<
+    boolean | undefined
+  >(undefined);
+  const { setPauseTimestamp } = useActivityTimer();
+
+  useEffect(() => {
+    checkBiometry();
+    checkBiometryInPreferences();
+  }, []);
+
+  let appListener: PluginListenerHandle;
+  useEffect(() => {
+    const updateBiometrics = async () => {
+      appListener = await BiometricAuth.addResumeListener(setBiometricInfo);
+      try {
+        appListener = await BiometricAuth.addResumeListener(setBiometricInfo);
+      } catch (error) {
+        if (error instanceof Error) {
+          // TODO: handle error
+        }
+      }
+    };
+    updateBiometrics();
+    return () => {
+      appListener?.remove();
+    };
+  }, []);
+
+  const checkBiometryInPreferences = async () => {
+    try {
+      const biometrics = await PreferencesStorage.get(
+        PreferencesKeys.APP_BIOMETRY
+      );
+      setBiometricsIsEnabled(biometrics.enabled as boolean);
+    } catch (e) {
+      // TODO: handle error
+    }
+  };
+  const checkBiometry = async () => {
+    try {
+      const biometricResult = await BiometricAuth.checkBiometry();
+      setBiometricInfo(biometricResult);
+      return biometricResult;
+    } catch (error) {
+      // TODO: error getting biometricInfo
+    }
+  };
+
+  const handleBiometricAuth = async (): Promise<boolean | BiometryError> => {
+    const biometricResult = await checkBiometry();
+    if (!biometricResult?.isAvailable) {
+      if (!biometricResult?.strongBiometryIsAvailable) {
+        return new BiometryError(
+          i18n.t("biometry.weakbiometry"),
```

Comment:
Question here, since you had the info.plist part that you just removed. These are just error strings right? And not user facing (as we'd never want to talk about info.plist to the user)

If so, these should not be in i18n - they should just be in English. Only user facing strings are i18n/translated.

Reply:
We should prompt an error notification for the end user in case the device biometry is too weak.
We can use this translated string once we have errors handling implemented

Reply:
Mixing error types and user facing error prompts is not a good practice as there could be other errors that will get thrown. We should throw errors with strings that are not in i18n.

And then in the UI where it's applicable, they can catch the error and handle it. If it's a completely different error they should get "Something went wrong"

It definitely doesn't make sense to show an error to the user that Info.plist wasn't setup properly.

Reply:
Okay, I understand the difference.
If the biometrics is weak, we don't even provide the option to configure the device with biometrics then this error would never show up.